rma_sale/views/sale_portal_template.xml

```diff
@@ -1,7 +1,11 @@
 <odoo>
     <template id="sale_order_portal_template" name="Request RMA" inherit_id="sale.sale_order_portal_template">
         <xpath expr="//div[hasclass('o_portal_sale_sidebar')]//div[hasclass('o_download_pdf')]/.." position="after">
-            <li t-if="sale_order.state in ['sale', 'done']" class="list-group-item flex-grow-1" id="li-request-rma">
+            <!-- If the sale order was not linked to the partner of the logged user, the button will not be shown -->
+            <li t-if="sale_order.state in ['sale', 'done'] and env.user.partner_id.commercial_partner_id == sale_order.partner_id.commercial_partner_id"
```

Comment:
I don't think we should restrict this: if you are able to see the sales orders due to the permissions, then we can request an RMA.

Reply:
Good point